henson_mongodb.py

```diff
@@ -0,0 +1,143 @@
+"""MongoDB plugin for Henson."""
+
+import os
+import pkg_resources
+import ssl
+
+from henson import Extension
+from motor.motor_asyncio import (
+    AsyncIOMotorClient,
+    AsyncIOMotorReplicaSetClient,
+)
+from pymongo import uri_parser
+
+__all__ = ('MongoDB',)
+
+try:
+    _dist = pkg_resources.get_distribution(__name__)
+    if not __file__.startswith(os.path.join(_dist.location, __name__)):
+        # Manually raise the exception if there is a distribution but
+        # it's installed from elsewhere.
+        raise pkg_resources.DistributionNotFound
+except pkg_resources.DistributionNotFound:
+    __version__ = 'development'
+else:
+    __version__ = _dist.version
+
+
+class MongoDB(Extension):
+    """An interface to interact with MongoDB."""
+
+    REQUIRED_SETTINGS = ('MONGODB_URI',)
+
+    DEFAULT_SETTINGS = {
+        'MONGODB_URI': None,
```

Comment:
Good catch. An earlier pass at this had it as optional with all of the individual pieces as their own settings as well.